### PR TITLE
docs: Add documentation on Spack builds

### PR DESCRIPTION
**Pull request by stephenswat**
This commit adds a new section to the "getting started" section of the documentation detailing how Spack can be used to install the dependencies of ACTS.

### CODE REVIEW

**Comment by paulgessinger** [comment]
Wondering if we should keep the part in getting started relatively short and link to a separate doc page with more info like how environments work. 

There we could also add documentation on the build caches.

File: docs/getting_started.md [hunk 1]
```diff
@@ -188,6 +188,60 @@ However, if you have the necessary prerequisites installed it is possible to use
...
+of ACTS you want to build. The final part is a series of variants, marked with
+the `+` symbol. The command above allows you to build virtually all components
+of ACTS, but it also pulls in a large number of dependencies. You can
+optionally remove variants you do not need in order to speed up the Spack
+build.
+
+Once Spack has concretized and installed the ACTS dependencies, you can build
+ACTS as normal using CMake.
+
+Spack support is experimental and not officially supported. Because ACTS' dependencies are constantly changing, it is possible that Spack's database may at times be out of date. If you encouter any problems using Spack to build ACTS, unofficial support is available in the #Spack channel on the ACTS Mattermost.
```
Comment (AJPfleger) [inline docs/getting_started.md:243]: Could we also add the link to the channel itself? :)

**Review by AJPfleger — APPROVED** [review]

File: docs/getting_started.md (hunk 1 above)
Comment (stephenswat) [reply docs/getting_started.md:243]: Done.

**Comment by stephenswat** [comment]
I've moved the Spack section to a new "Miscellaneous topics" section and linked to it.

**Review by paulgessinger — APPROVED** [review]

**Comment by andiwand** [comment]
What I am missing a bit from the documentation is how do we sync spack if it gets outdated? If I understand correctly from a previous discussion they keep all the descriptions in the main repository and we would have to update there? Is there another way? Can we host the recipe somewhere?

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [d7392a5ab856e296ab9e4b33f61324aa367891a0]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [report_pull_request](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822208)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822207)||
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822206)||
|:green_circle: | [test_ActsDumpGeometryIdentifiers](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822205)||
|:green_circle: | [test_ActsCheckObjectCountsCachedWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822204)||
|:green_circle: | [test_ActsCheckObjectCountsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822203)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822202)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822201)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822200)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822199)||
|:green_circle: | [test_ActsWorkflowHeavyIons](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822198)||
|:green_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822197)||
|:green_circle: | [test_ActsWorkflowCached](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822196)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822195)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822194)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822193)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822192)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822191)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822190)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822189)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822188)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822187)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822186)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822185)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822184)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822183)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822182)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822181)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822180)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822179)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822178)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822177)||
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42822176)||

**Pull request merged**